tools/sense_studio/static/testing.js

```diff
@@ -0,0 +1,120 @@
+
+function addTerminalMessage(message) {
+    let terminal = document.getElementById('testTerminal');
+    terminal.insertAdjacentHTML('beforeend', `<div class='monospace-font'><b>${message}</b></div>`);
+    terminal.scrollTop = terminal.scrollHeight;
+}
+
+function streamVideo(message) {
+    let frame = document.getElementById('frame');
+    frame.src = message.image;
+}
+
+async function getPathInputs(pathLabelId, pathInputId) {
+    let pathLabel = document.getElementById(pathLabelId);
+    let pathInput = document.getElementById(pathInputId);
+    let project = document.getElementById('project');
+
+    let name = project.value;
+    let path = pathInput.value;
+
+    let directoriesResponse = await browseDirectory(path, name);
+    let disabled = false;
+
+    // Check that project path is filled and exists
+    if (path === '') {
+        setFormWarning(pathLabel, pathInput, '');
+        disabled = true;
+    } else if (!directoriesResponse.path_exists) {
+        setFormWarning(pathLabel, pathInput, 'This path does not exist');
+        disabled = true;
+    } else {
+        setFormWarning(pathLabel, pathInput, '');
+    }
+}
+
+async function startTesting(url){
+    let classifier = document.getElementById('classifier').value;
+    let inputVideoPath = document.getElementById('inputVideoPath').value;
+    let outputVideoName = document.getElementById('outputVideoName').value;
+    let path = document.getElementById('path').value;
+    let title = document.getElementById('title').value;
+    let buttonTest = document.getElementById('btnTest');
+    let buttonCancelTest = document.getElementById('btnCancelTest');
+    let video_stream = document.getElementById('videoStream');
+
+    data = {
+        classifier: classifier,
+        inputVideoPath: inputVideoPath,
+        outputVideoName: outputVideoName,
+        path: path,
+        title: title,
+    };
+
+    buttonTest.disabled = true;
+    buttonCancelTest.disabled = false;
+
+    await asyncRequest(url, data);
+
+    let socket = io.connect('/stream-video');
+    socket.on('connect', function() {
+        console.log('Socket Connected');
+        socket.emit('stream_video', {status: 'Socket Connected'});
+    });
+
+    socket.on('testing_images', function(message) {
+        streamVideo(message);
+    });
+
+    socket.on('success', function(message) {
+        if (message.status === 'Complete') {
+            addTerminalMessage('Stopping Inference...');
```

Comment:
This message only shows up on the terminal a few seconds after pressing cancel when everything else is already stopped, so it's a little useless.